src/components/DiaryCard/AddNewDiary.js

```diff
@@ -1,67 +1,63 @@
 import Card from "@material-ui/core/Card";
+import DiaryCard from "./DiaryCard";
 
 
-function AddNewDiary(props) {
+function AddNewDiary() {
 
     //states
     const [diaryData, setDiaryData] = useState({
         title: '',
-        author: '',
         description: ''
     })
 
+    const [isShowCard,setIsShowCard]=useState(false);
 
-    function sendDiaryDataHandler() {
-        console.log(diaryData);
-        setDiaryData({
-            title: "",
-            author: "",
-            description: ""
-        })
-    }
+
+    const allData=[{
+        title:diaryData.title,
+        description:diaryData.description
+    }]
 
 
-    function submitDataHandler(event) {
-        event.preventDefault();
+    function submitDataHandler() {
+        if (diaryData.title===''){
+            console.log("Title Missing")
+        }else if(diaryData.description===''){
+            console.log('Description Missing')
+        }else {
+            setIsShowCard(true);
+        }
+        console.log(allData)
     }
 
-    return (
-        <Card className={classes.card}>
-            <form className={classes.form} onSubmit={submitDataHandler}>
-                <div className={classes.control}>
-                    <label htmlFor='title'>
-                        Topic
-                    </label>
-                    <input placeholder='Topic' type='text' value={diaryData.title} required id='title' onChange={(e) => {
-                        setDiaryData(prevState => ({...prevState, title: e.target.value}))
-                    }}/>
-                </div>
-                <div className={classes.control}>
-                    <label htmlFor='author'>
-                        Subtopic
-                    </label>
-                    <input placeholder='Subtopic' type='text'  value={diaryData.author} required id='author' onChange={(e) => {
-                        setDiaryData(prevState => ({...prevState, author: e.target.value}))
-                    }}/>
-                </div>
-                <div className={classes.control}>
-                    <label htmlFor='description'>
-                        Description
-                    </label>
-                    <textarea placeholder='Description' id='description'  value={diaryData.description} required rows='5' onChange={(e) => {
-                        setDiaryData(prevState => ({...prevState, description: e.target.value}))
-                    }}>
+    function preventHandler(event) {
+        event.preventDefault()
+    }
 
+    return (
+        <div>
+            <Card className={classes.card}>
+                <form className={classes.form} onSubmit={preventHandler}>
+                    <div className={classes.control}>
+                        <label htmlFor='title'>Topic</label>
+                        <input placeholder='Topic' type='text' value={diaryData.title} required id='title' onChange={(e) => {setDiaryData(prevState => ({...prevState, title: e.target.value}))}}/>
```

Comment:
Declare a function for this onChange and assign it to here. Using inline like this is not a good practice.

Reply:
I have Declare a function Instead of inline code